wire-library/wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt

```diff
@@ -971,6 +971,87 @@ class KotlinGeneratorTest {
         repoBuilder.generateGrpcKotlin("routeguide.RouteGuide", "RouteChat"))
   }
 
+  @Test fun nameSuffixes() {
+    //language=kotlin
+    val suspendingInterface = """
+          |package routeguide
+          |
+          |import com.squareup.wire.Service
+          |import com.squareup.wire.WireRpc
+          |
+          |/**
+          | * RouteGuide service interface.
+          | */
+          |public interface RouteGuide : Service {
+          |  /**
+          |   * Returns the \[Feature\] for a \[Point\].
+          |   */
+          |  @WireRpc(
+          |    path = "/routeguide.RouteGuide/GetFeature",
+          |    requestAdapter = "routeguide.Point#ADAPTER",
+          |    responseAdapter = "routeguide.Feature#ADAPTER",
+          |    sourceFile = "routeguide.proto"
+          |  )
+          |  public suspend fun GetFeature(request: Point): Feature
+          |}
+          |""".trimMargin()
+
+    //language=kotlin
+    val blockingInterface = """
+          |package routeguide
+          |
+          |import com.squareup.wire.Service
+          |import com.squareup.wire.WireRpc
+          |
+          |/**
+          | * RouteGuide service interface.
+          | */
+          |public interface RouteGuideBlockingThing : Service {
```

Comment:
So the word `Blocking` shows up here and I think it shouldn’t? Like because the user’s requested suffix is `Client` and not `BlockingClient` I think this interface should be `RouteGuideClient` ?

Reply:
So you're thinking that the presence of `nameSuffix` overrides not just "Client"/"Server", but also the other suffix components like "Blocking"?  I hadn't considered that, but it seems makes intuitive sense.  Let me know if I understand you properly - it should be an easy fix.

Reply:
Yep exactly!

Reply:
Cool; done!